machine/v1beta1/types_azureprovider.go

```diff
@@ -397,6 +419,28 @@ type OSDiskManagedDiskParameters struct {
 	// DiskEncryptionSet is the disk encryption set properties
 	// +optional
 	DiskEncryptionSet *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+	// SecurityProfile specifies the security profile for the managed disk.
+	// +optional
+	SecurityProfile *VMDiskSecurityProfile `json:"securityProfile,omitempty"`
+}
+
+// VMDiskSecurityProfile specifies the security profile settings for the managed disk.
+// It can be set only for Confidential VMs.
+type VMDiskSecurityProfile struct {
+	// DiskEncryptionSet specifies the customer managed disk encryption set resource id for the
+	// managed disk that is used for Customer Managed Key encrypted ConfidentialVM OS Disk and
+	// VMGuest blob.
+	// +optional
+	DiskEncryptionSet *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+	// SecurityEncryptionType specifies the encryption type of the managed disk.
+	// It is set to DiskWithVMGuestState to encrypt the managed disk along with the VMGuestState
+	// blob, and to VMGuestStateOnly to encrypt the VMGuestState blob only.
+	// When set to VMGuestStateOnly, the vTPM should be enabled.
+	// When set to DiskWithVMGuestState, both SecureBoot and vTPM should be enabled.
+	// It can be set only for Confidential VMs.
```

Comment:
How do I make this a confidential VM?

Reply:
Azure Confidential VMs are defined by their `SecurityProfile.SecurityType` being set to `ConfidentialVM` and the `SecurityEncryptionType` being set to one the allowed values `VMGuestStateOnly` and `DiskWithVMGuestState`, for the supported VM sizes [DCasv5 and ECasv5](https://learn.microsoft.com/en-us/azure/confidential-computing/confidential-vm-overview).

Reply:
Might be worth explaining this here as a relative to the providerSpec, so that's it's captured in the godoc, I probably wouldn't know where I'm setting this otherwise